### DISCUSSION TITLE
Where does Colima obtain the username for the Docker daemon and what relevance does it have?

### DISCUSSION THREAD

**Discussion by bartek** [abiosoft/colima#699] [Q&A]
I'm having permission issues which seem related to Colima, as I am not getting these issues when I use Docker desktop, skipping the Colima runtime.

When I run `docker info | grep Username`, I see that my username listed is `bartekci`. However, my $USER is `bartek`. The referenced username in docker info here seems related to my Docker hub login, so I'm not sure if this is related in any way.

The problem I seem to be having is that when I run a particular container (run as a container user named `vespa`), I am having permission denied issues in that container cannot write to a mounted volume. The host directory has ownership `bartek:staff`, which is what I'd expect.

When I run the same `vespa` container with Docker desktop, I have no permission issues.

I'm trying to figure out how I debug this and what I should be looking at, but I'm not sure where to start. Can anyone help?

**Comment by abiosoft** [comment]
Do you rename your user or your $HOME directory uses `bartekci`? 

Can you share the exact errors you're getting with teh volume mounts?

**Comment by bartek** [comment]
@abiosoft My $HOME is using `bartek`. There is no reference to `bartekci` in my machine except from `docker info`. Is my understanding correct that this is in relation to Docker Hub authentication? Does it have any relevance to the daemon?

The particular error is not coming from Colima but from the container. We can see:

```
Attaching to lumen-vespa-1
lumen-vespa-1  | 1681923196.922806	8f791057a48f	10	-	vespa-start-configserver	warning	Could not create a file in directory 'logs'
lumen-vespa-1  | 1681923196.922833	8f791057a48f	10	-	vespa-start-configserver	warning	problem: open logs/tmp.probe.1724060353.tmp: permission denied
lumen-vespa-1  | 1681923196.922836	8f791057a48f	10	-	vespa-start-configserver	warning	Currently running as user: vespa
lumen-vespa-1  | 1681923196.922838	8f791057a48f	10	-	vespa-start-configserver	warning	Wanted logs to be owned by user id: 1000
lumen-vespa-1  | 1681923196.922839	8f791057a48f	10	-	vespa-start-configserver	warning	Wanted logs to have group id: 1000
lumen-vespa-1  | 1681923196.922842	8f791057a48f	10	-	vespa-start-configserver	warning	Wanted logs to have permissions: -rwxr-xr-x
lumen-vespa-1  | 1681923196.922844	8f791057a48f	10	-	vespa-start-configserver	warning	current status of logs is:
lumen-vespa-1  | 1681923196.926176	8f791057a48f	10	-	vespa-start-configserver	warning	 File: logs
```

We can see that the container expects user 1000/1000. On Ubuntu, this is typically `ubuntu` user. On macOS, this id does not seem to exist. Regardless, I chown the directory to `1000:1000` but I still have write issues as shown above (same error).

I'm concerned there is something fundamental I am missing here. It's a bit opaque to me in that I'm not clear about the user/group/permissions story in relation to this container and the mount it is trying to write to.

**Comment by abiosoft** [comment]
Can you share the docker image or docker compose file, for me to reproduce it?

Yeah, the user 1000 does not exist on macOS, it should generally work unless the container is trying to `chmod` the files/folders. 
Can you also share the output of `colima status`?

**Reply by bartek** [reply]
@abiosoft Appreciate it! Here's more details:

> it should generally work unless the container is trying to chmod the files/folders.

I believe this is what is happening. On startup there is the attempt to change ownership.

My docker-compose entry:

```
version: '3.7'
services:
  vespa:
    image: vespaengine/vespa:8.153.19
    volumes:
      - /opt/lumen/vespa/var:/opt/vespa/var
      - /opt/lumen/vespa/logs:/opt/vespa/logs
    restart: on-failure
    ports:
      # Doc and query APIs
      - 443:443 # tls
      - 8080:8080 # document API
      - 19071:19071 # config API
```

Mounts:

```
❯ ls -al /opt/lumen
total 0
drwxrwxrwx  4 1000  1000   128 18 Apr 20:58 .
drwxr-xr-x  4 root  wheel  128 17 Jan 11:50 ..
drwxrwxrwx  2 1000  1000    64 17 Jan 11:50 opensearch
drwxrwxrwx  4 1000  1000   128 20 Apr 06:47 vespa
```

The Dockerfile of this `vespa` service is here: https://github.com/vespa-engine/docker-image/blob/master/Dockerfile

And `colima status`

```
❯ colima status
INFO[0000] colima is running using QEMU                 
INFO[0000] arch: aarch64                                
INFO[0000] runtime: docker                              
INFO[0000] mountType: sshfs                             
INFO[0000] socket: unix:///Users/bartek/.colima/default/docker.sock 
```
